sql/idempotent/01-openai.sql

```diff
@@ -0,0 +1,215 @@
+
```

Comment:
Not reviewing this as I am assuming no change, just moving code around? (everything in dir)

Reply:
I did move some code from the bodies of the SQL functions to the Python module